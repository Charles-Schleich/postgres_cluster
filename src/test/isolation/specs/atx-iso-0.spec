# The main xact in s1 should see (1, 1), (2, 0) if "s2c" is done before "s1s",
# and (1, 0), (2, 0) otherwise.

setup
{
  create table t(a int, b int);
  insert into t values (1, 1), (2, 2);
}

teardown
{
  drop table t;
}

session "s1"
setup         { begin isolation level read committed; }
step "s1atxb" { begin autonomous; }
step "s1atxu" { update t set b = 0 where a = 2; }
step "s1atxc" { commit; }
step "s1s"    { select * from t; }
step "s1c"    { commit; }

session "s2"
setup      { begin; }
step "s2u" { update t set b = 0 where a = 1; }
step "s2c" { commit; }
