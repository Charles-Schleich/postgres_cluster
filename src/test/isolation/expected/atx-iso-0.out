Parsed test spec with 2 sessions

starting permutation: s1atxb s1atxu s1atxc s1s s1c s2u s2c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s1c: commit;
step s2u: update t set b = 0 where a = 1;
step s2c: commit;

starting permutation: s1atxb s1atxu s1atxc s1s s2u s1c s2c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2u: update t set b = 0 where a = 1;
step s1c: commit;
step s2c: commit;

starting permutation: s1atxb s1atxu s1atxc s1s s2u s2c s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2u: update t set b = 0 where a = 1;
step s2c: commit;
step s1c: commit;

starting permutation: s1atxb s1atxu s1atxc s2u s1s s1c s2c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s2u: update t set b = 0 where a = 1;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s1c: commit;
step s2c: commit;

starting permutation: s1atxb s1atxu s1atxc s2u s1s s2c s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s2u: update t set b = 0 where a = 1;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2c: commit;
step s1c: commit;

starting permutation: s1atxb s1atxu s1atxc s2u s2c s1s s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s2u: update t set b = 0 where a = 1;
step s2c: commit;
step s1s: select * from t;
a              b              

2              0              
1              0              
step s1c: commit;

starting permutation: s1atxb s1atxu s2u s1atxc s1s s1c s2c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s2u: update t set b = 0 where a = 1;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s1c: commit;
step s2c: commit;

starting permutation: s1atxb s1atxu s2u s1atxc s1s s2c s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s2u: update t set b = 0 where a = 1;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2c: commit;
step s1c: commit;

starting permutation: s1atxb s1atxu s2u s1atxc s2c s1s s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s2u: update t set b = 0 where a = 1;
step s1atxc: commit;
step s2c: commit;
step s1s: select * from t;
a              b              

2              0              
1              0              
step s1c: commit;

starting permutation: s1atxb s1atxu s2u s2c s1atxc s1s s1c
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s2u: update t set b = 0 where a = 1;
step s2c: commit;
step s1atxc: commit;
step s1s: select * from t;
a              b              

2              0              
1              0              
step s1c: commit;

starting permutation: s1atxb s2u s1atxu s1atxc s1s s1c s2c
step s1atxb: begin autonomous;
step s2u: update t set b = 0 where a = 1;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s1c: commit;
step s2c: commit;

starting permutation: s1atxb s2u s1atxu s1atxc s1s s2c s1c
step s1atxb: begin autonomous;
step s2u: update t set b = 0 where a = 1;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2c: commit;
step s1c: commit;

starting permutation: s1atxb s2u s1atxu s1atxc s2c s1s s1c
step s1atxb: begin autonomous;
step s2u: update t set b = 0 where a = 1;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s2c: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s1atxb s2u s1atxu s2c s1atxc s1s s1c
step s1atxb: begin autonomous;
step s2u: update t set b = 0 where a = 1;
step s1atxu: update t set b = 0 where a = 2;
step s2c: commit;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s1atxb s2u s2c s1atxu s1atxc s1s s1c
step s1atxb: begin autonomous;
step s2u: update t set b = 0 where a = 1;
step s2c: commit;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s2u s1atxb s1atxu s1atxc s1s s1c s2c
step s2u: update t set b = 0 where a = 1;
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s1c: commit;
step s2c: commit;

starting permutation: s2u s1atxb s1atxu s1atxc s1s s2c s1c
step s2u: update t set b = 0 where a = 1;
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              1              
2              0              
step s2c: commit;
step s1c: commit;

starting permutation: s2u s1atxb s1atxu s1atxc s2c s1s s1c
step s2u: update t set b = 0 where a = 1;
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s2c: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s2u s1atxb s1atxu s2c s1atxc s1s s1c
step s2u: update t set b = 0 where a = 1;
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s2c: commit;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s2u s1atxb s2c s1atxu s1atxc s1s s1c
step s2u: update t set b = 0 where a = 1;
step s1atxb: begin autonomous;
step s2c: commit;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;

starting permutation: s2u s2c s1atxb s1atxu s1atxc s1s s1c
step s2u: update t set b = 0 where a = 1;
step s2c: commit;
step s1atxb: begin autonomous;
step s1atxu: update t set b = 0 where a = 2;
step s1atxc: commit;
step s1s: select * from t;
a              b              

1              0              
2              0              
step s1c: commit;
